src/languages/erlang.js

```diff
@@ -80,9 +80,39 @@ export default function(hljs) {
     scope: 'string',
     match: /\$(\\([^0-9]|[0-9]{1,3}|)|.)/,
   };
+  const TRIPLE_QUOTE = hljs.END_SAME_AS_BEGIN({
+    scope: 'string',
+    begin: /("{3,})/,
+    end: /("{3,})/,
+  });
+
+  // the closing character depends on the kind of opening character.
+  const SIGIL_START_END_PAIRS = {
+    '(': ')',
+    '[': ']',
+    '{': '}',
+    '<': '>',
+  };
+  const SIGIL = {
+    scope: 'string',
+    begin: /~\w?("{3,}|[\(\[\{<\/|'"`#])/,
+    end: /("{3,}|[\)\]\}>\/|'"`#])/,
```

Comment:
This looks like it could have expensive false positives in other languages (auto-detect), so we might need to disable auto-detect for Erlang if we go this direction...

Reply:
That actually might help for languages that use `~` to define variables, etc... but is Erlang the only lisp that uses `~[bracket]` style patterns or would that apply to most LISPs?

Reply:
There is some more technical information in the feature proposal [EEP 66](https://www.erlang.org/eeps/eep-0066).
It's based on previous work in Elixir (and that's a more popular Erlang-like so you may find information on how other tools deal with this online).

Note that some ideas in that proposal did not make it into the release, namely the `r` prefix for regular expressions.

Reply:
What if we just used individual variants here instead of this matcher?

Reply:
I thought about that as well. How would I do that in a way that fits the highlight.js code style?
Something like this?
```js
const SIGILS = [
  { ... },
  { ... },
];
// and then instead of SIGIL use
   ...SIGILS
```

Reply:
We also have the literal `variants` syntax, but I think `...SIGILS` is fine (as variants just gets expanded to the same thing anyways really).

Reply:
`variants` is exactly what I was looking for :)